rollup.config.js

```diff
@@ -0,0 +1,62 @@
+import nodeResolve from "@rollup/plugin-node-resolve";
+import commonjs from "@rollup/plugin-commonjs";
+import babel from "@rollup/plugin-babel";
+const createEsm = process.env.CREATE_ESM === "true";
+const createCjs = process.env.CREATE_CJS === "true";
+const createUmd = process.env.CREATE_UMD === "true";
+const input = "src/index.js"; // Update with your entry point
+const outputDir = "build"; // Update with your desired output directory
+const name = "Cytposcape";
+const configs = [];
+
+if (createEsm) {
+  configs.push({
+    input,
+    output: {
+      file: `${outputDir}/cytoscape.esm.js`,
+      format: "es",
+    },
+    plugins: [
+      nodeResolve(),
+      commonjs(),
+      babel({
+        exclude: "node_modules/**",
+      }),
+    ],
+  });
+}
+if (createCjs) {
+  configs.push({
+    input,
+    output: {
+      file: `${outputDir}/cytoscape.cjs.js`,
+      format: "cjs",
+    },
+    plugins: [
+      nodeResolve(),
+      commonjs(),
+      babel({
+        exclude: "node_modules/**",
+      }),
+    ],
+  });
+}
+if (createUmd) {
+  configs.push({
+    input,
+    output: {
+      file: `${outputDir}/cytoscape.umd.js`,
+      format: "umd",
+      name: name,
+    },
+    plugins: [
+      nodeResolve(),
+      commonjs(),
+      babel({
+        exclude: "node_modules/**",
+      }),
+    ],
```

Comment:
It looks like you’re excluding the dependencies (node_modules) so that users can manage the versions of the dependencies themselves if needed.  That’s good.

It would be useful to also have two other builds:

(1) cytoscape-automove.esm.min.js — ESM build with all dependencies bundled and the entire file minified

(2) cytoscape-automove.umd.min.js — UMD build with all dependencies bundled and the entire file minified

Both (1) and (2) are useful for use cases where the user wants to have a plain HTML+JS page without a build system.  For that, it’s nice to have a single file that ‘just works’ by copying it into a folder.

Reply:
Ok, I did some digging, looks like i have to install the dev dependency called terser, and just add one more push func to the configs array, but it does not require and special value for the format parameter, it's just umd. Will that suffice?

Reply:
Terser should work with all configs. Sounds good

Reply:
I have updated the PR, please check

Reply:
If everything is fine, please assign the issue to me, on cytoscape's main repo.